src/scarb.rs

```diff
@@ -0,0 +1,119 @@
+use std::{env::current_dir, fs, path::PathBuf};
+
+use anyhow::Context;
+use cairo_lang_test_plugin::TestCompilation;
+use cairo_lang_test_runner::{CompiledTestRunner, TestRunConfig};
+use camino::Utf8PathBuf;
+
+use scarb::{
+    core::{Config, TargetKind},
+    ops::{self, collect_metadata, CompileOpts, MetadataOptions},
+};
+
+pub fn prepare_crate_for_exercise(file_path: &PathBuf) -> PathBuf {
+    let crate_path = current_dir().unwrap().join(PathBuf::from("runner-crate"));
+    let src_dir = crate_path.join("src");
+    if !src_dir.exists() {
+        let _ = fs::create_dir(&src_dir);
+    }
+    let lib_path = src_dir.join("lib.cairo");
+    let file_path = current_dir().unwrap().join(file_path);
+
+    match fs::copy(&file_path, &lib_path) {
+        Ok(_) => {}
+        Err(err) => panic!("Error occurred while preparing the exercise,\nExercise: {file_path:?}\nLib path: {lib_path:?}\n{err:?}"),
+    };
+    crate_path
+}
+
+pub fn scarb_build(file_path: &PathBuf) -> anyhow::Result<String> {
+    let crate_path = prepare_crate_for_exercise(file_path);
+    let config = scarb_config(crate_path);
+
+    match compile(&config, false) {
+        Ok(_) => Ok("".into()),
+        Err(_) => anyhow::bail!("Couldn't build the exercise..."),
+    }
+}
+
+pub fn scarb_test(file_path: &PathBuf) -> anyhow::Result<String> {
```

Comment:
Would love to have a quick line of documentation about what you're doing here with runner-crate, etc - just so that it's maintainable in the future.

Reply:
Aye sure.

Reply:
This was really helpful. And everything worked. But I had a little bug, and I had updated the path to fix that 😂 
It's all smooth now. 💋 
Thanks!